src/libraries/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.SslCtx.cs

```diff
@@ -69,7 +83,107 @@ protected override bool ReleaseHandle()
         {
             Interop.Ssl.SslCtxDestroy(handle);
             SetHandle(IntPtr.Zero);
+            if (_gch.IsAllocated)
+            {
+                //Interop.Ssl.SslCtxSetData(this, (IntPtr)_gch);
+                _gch.Free();
+            }
+
+            if (_sslSessions != null)
+            {
+                lock (_sslSessions)
+                {
+                    foreach (string name in _sslSessions.Keys)
+                    {
+                        _sslSessions.Remove(name, out IntPtr session);
+                        Interop.Ssl.SessionFree(session);
+                    }
+                }
+            }
+
             return true;
         }
+
+        public void EnableSessionCache()
```

Comment:
We don't have a consistent rule here, but personally I wish that it was a compile error to declare a public method inside a non-public class.  I'd declare all of these new methods as internal.